docs/pages/_error.js

```diff
@@ -93,6 +93,11 @@ export default class Error extends React.Component {
       redirectPath = `https://reactnative.dev/docs/${page}`;
     }
 
+    // Remove versiong from path if the version is still supported, to redirect to the root
+    if (isVersionedPath(redirectPath) && isVersionDocumented(redirectPath)) {
```

Comment:
This should be the last resort solution. It redirects users back to the root of the versioned pages when the version is still supported. This allows us to "blindly" reuse the versioned path when switching versions, and make sure the user either reaches the same page or is redirected to root (like the current behavior)